daly-bms-uart.cpp

```diff
@@ -174,10 +389,12 @@ bool Daly_BMS_UART::validateChecksum()
 
 void Daly_BMS_UART::barfRXBuffer(void)
 {
-    Serial.printf("<DALY-BMS DEBUG> RX Buffer: [");
+#ifdef DALY_BMS_DEBUG
+    DEBUG_SERIAL.print("<DALY-BMS DEBUG> RX Buffer: [");
     for (int i = 0; i < XFER_BUFFER_LENGTH; i++)
     {
-        Serial.printf("0x%x,", this->my_rxBuffer[i]);
+        DEBUG_SERIAL.print("0x" + (String)this->my_rxBuffer[i]);
```

Comment:
```suggestion
        DEBUG_SERIAL.print(",0x");
        DEBUG_SERIAL.print(this->my_rxBuffer[i], HEX);
```
see https://www.arduino.cc/reference/en/language/functions/communication/serial/print/

the leading comma isn't ideal formatting, but it's better then no commas in my opinion

Reply:
yes, i have modified it again, update later